src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -193,7 +193,40 @@ private static void configureSslContext(HttpClientConfig.Builder builder,
                                             ElasticsearchSinkConnectorConfig config) {
     SslFactory kafkaSslFactory = new SslFactory(Mode.CLIENT, null, false);
     kafkaSslFactory.configure(config.sslConfigs());
-    SSLContext sslContext = kafkaSslFactory.sslContext();
+
+    SSLContext sslContext;
+    try {
+      // try AK <= 2.2 first
+      sslContext =
+          (SSLContext) SslFactory.class.getDeclaredMethod("sslContext").invoke(kafkaSslFactory);
+      log.trace("Using AK 2.2< SslFactory.");
+    } catch (Exception e) {
+      // must be running AK 2.3+
+      log.trace("Could not find SslFactory::sslContext. Trying AK 2.3+ APIs.");
+
+      Object ss;
```

Comment:
maybe (for better readability): 

```suggestion
      Object sslEngine;
```